### DISCUSSION TITLE
Does TSFEL work with GPUs?

### DISCUSSION THREAD

**Discussion by manelbenabid** [fraunhoferportugal/tsfel#117] [Q&A]
Hello,

I am using tsfel to extract features from my dataset but it is taking a very long time. Does the library support GPUs or should I wait for the process to finish?

Thank you!

**Comment by dmfolgado** [comment]
Hi,

TSFEL does not support GPU acceleration. Nevertheless, it supports multiprocessing on the CPU. You can enable it by setting `n_jobs=-1` as input parameter of the [`time_series_features_extractor`](https://tsfel.readthedocs.io/en/latest/descriptions/modules/tsfel.feature_extraction.html#tsfel.feature_extraction.calc_features.time_series_features_extractor) method.



Let us know if it did improve.
